tests/chainer_tests/functions_tests/normalization_tests/test_batch_renormalization.py

```diff
@@ -1,218 +1,268 @@
-        self.dmax = self.dtype(5)
+        shape = (5, 3) + (2,) * self.ndim
+        aggr_shape = (3,)
+        self.running_mean = (
+            numpy.random.uniform(-1, 1, aggr_shape).astype(self.dtype))
+        self.running_var = (
+            numpy.random.uniform(1e-3, 1, aggr_shape).astype(self.dtype))
 
-        self.gamma = numpy.random.uniform(.5, 1, (3,)).astype(self.dtype)
-        self.beta = numpy.random.uniform(-1, 1, (3,)).astype(self.dtype)
+        axis = (0,) + tuple(six.moves.range(2, self.ndim + 2))
+        expander = (None, Ellipsis) + (None,) * self.ndim
 
-        shape = (5, 3) + (2,) * self.ndim
-        self.x = numpy.random.uniform(-1, 1, shape).astype(self.dtype)
-        self.gy = numpy.random.uniform(-1, 1, shape).astype(self.dtype)
-
-        self.args = [self.x, self.gamma, self.beta]
-        self.mean = self.x.mean(axis=self.aggr_axes)
-        self.var = self.x.var(axis=self.aggr_axes) + self.eps
-        # Need to add some noise to running_mean and running_var,
-        # otherwise we will always get r=1, d=0
-        self.running_mean = self.mean + numpy.random.uniform(
-            -1, 1, self.mean.shape).astype(self.dtype)
-        self.running_var = numpy.abs(self.var + numpy.random.uniform(
-            -1, 1, self.var.shape).astype(self.dtype))
-
-        self.train = True
-        self.check_forward_options = {'atol': 1e-4, 'rtol': 1e-3}
-        self.check_backward_options = {'atol': 1e-4, 'rtol': 1e-3}
         if self.dtype == numpy.float16:
-            self.check_forward_options = {'atol': 1e-3, 'rtol': 1e-2}
-            self.check_backward_options = {'atol': 1e-3, 'rtol': 1e-2}
-
-    def check_compare_naive(self, args, stats, y_grad):
-        def compute(f):
-            x, gamma, beta = [chainer.Variable(v.copy()) for v in args]
-            running_mean, running_var = [v.copy() for v in stats]
-            y = f(x, gamma, beta, running_mean, running_var)
-            y.grad = y_grad.copy()
+            self.check_forward_options = {'atol': 0.02, 'rtol': 0.02}
+
+        self.shape = shape
+        self.aggr_shape = aggr_shape
+        self.axis = axis
+        self.expander = expander
+
+    def generate_inputs(self):
+        shape = self.shape
+        aggr_shape = self.aggr_shape
+        dtype = self.dtype
+
+        x = numpy.random.uniform(-10, 10, shape).astype(dtype)
+        gamma = numpy.random.uniform(.5, 1, aggr_shape).astype(dtype)
+        beta = numpy.random.uniform(-1, 1, aggr_shape).astype(dtype)
+        gy = numpy.random.uniform(-1, 1, shape).astype(dtype)
+        return x, gamma, beta, gy
+
+    def _compute_backward(self, x, gamma, beta, y, gy):
+        assert isinstance(x, chainer.Variable)
+        assert isinstance(gamma, chainer.Variable)
+        assert isinstance(beta, chainer.Variable)
+        assert isinstance(y, chainer.Variable)
+        assert isinstance(gy, chainer.Variable)
+
+        import chainerx
+        if isinstance(x.array, chainerx.ndarray):
```

Comment:
A detail but is this a "cleaner" way to check?
```suggestion
        if x.xp is chainerx:
```